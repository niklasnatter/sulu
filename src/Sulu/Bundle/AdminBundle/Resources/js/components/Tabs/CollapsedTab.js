// @flow
import React from 'react';
import collapsedTabStyles from './collapsedTab.scss';

type Props = {
    children: string,
    index: number,
    onClick: (index: number) => void,
};

export default class CollapsedTab extends React.PureComponent<Props> {
    handleClick = () => {
        const {
            index,
            onClick,
        } = this.props;

        onClick(index);
    };

    render() {
        const {
            children,
        } = this.props;

        return (
            <li className={collapsedTabStyles.collapsedTab}>
                <button
                    onClick={this.handleClick}
                    title={children}
                    type="button"
                >
                    {children}
                </button>
            </li>
        );
    }
}
