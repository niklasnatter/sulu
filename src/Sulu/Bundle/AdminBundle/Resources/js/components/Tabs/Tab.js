// @flow
import React from 'react';
import classNames from 'classnames';
import tabStyles from './tab.scss';
import type {Element, ElementRef} from 'react';
import type {Type} from './types';

type Props = {
    badges: Element<*>[],
    children: string,
    hidden: boolean,
    index?: number,
    onClick?: (index: ?number) => void,
    selected: boolean,
    tabRef?: (index: ?number, ref: ?ElementRef<'li'>) => void,
    type?: Type,
};

class Tab extends React.PureComponent<Props> {
    static defaultProps = {
        badges: [],
        hidden: false,
        selected: false,
    };

    setTabRef = (ref: ?ElementRef<'li'>) => {
        const {index, tabRef} = this.props;

        if (tabRef) {
            tabRef(index, ref);
        }
    };

    handleClick = () => {
        const {index, onClick} = this.props;

        if (onClick) {
            onClick(index);
        }
    };

    render() {
        const {
            badges,
            children,
            hidden,
            type,
            selected,
        } = this.props;

        const tabClass = classNames(
            tabStyles.tab,
            tabStyles[type],
            {
                [tabStyles.hidden]: hidden,
                [tabStyles.selected]: selected,
            }
        );

        return (
            <li className={tabClass} ref={this.setTabRef}>
                <button
                    disabled={selected}
                    onClick={this.handleClick}
                    title={children}
                    type="button"
                >
                    {children}
                    {!!badges && !!badges.length &&
                        <div className={tabStyles.badges}>
                            {badges}
                        </div>
                    }
                </button>
            </li>
        );
    }
}

export default Tab;
