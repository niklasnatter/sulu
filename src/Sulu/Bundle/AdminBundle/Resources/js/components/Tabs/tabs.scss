@import '../../containers/Application/colors.scss';
@import './variables.scss';

.tabs {
    display: flex;
    background-color: $tabMenuBackgroundColor;
    padding: 0 40px;

    &.inline {
        padding: 0 20px;
        background-color: $tabMenuTransparentBackgroundColor;
    }

    &.nested {
        border-radius: 3px 3px 0 0;
        padding: 0;
    }
}

.tabs-container-wrapper {
    flex: 1 1 auto;
    position: relative;
    height: $tabMenuHeight;
    overflow: hidden;
}

.button {
    flex: 0 0 auto;
    border: none;
    outline: none;
    background: none;
    box-shadow: none;
    cursor: pointer;
    padding: 10px;
}

.tabs-container {
    position: absolute;
    display: flex;
    list-style: none;
    margin: 0;
    padding: 0;

    & > li {
        display: block;
    }
}
