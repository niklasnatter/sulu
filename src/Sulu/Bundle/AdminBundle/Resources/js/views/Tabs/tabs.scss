@import '../../containers/Application/variables.scss';

.tabs-container {
    margin: -$viewPaddingVertical -$viewPaddingHorizontal $viewPaddingVertical;

    &.nested {
        margin: 0 0 $viewPaddingVertical;
    }

    &.disable-gap {
        margin: 0;
    }
}
