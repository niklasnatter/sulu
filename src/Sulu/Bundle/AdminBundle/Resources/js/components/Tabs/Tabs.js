// @flow
import React from 'react';
import {action, computed, observable} from 'mobx';
import {observer} from 'mobx-react';
import classNames from 'classnames';
import debounce from 'debounce';
import Popover from '../Popover';
import Icon from '../Icon';
import Tab from './Tab';
import CollapsedTabList from './CollapsedTabList';
import CollapsedTab from './CollapsedTab';
import tabsStyles from './tabs.scss';
import type {Type} from './types';
import type {ChildrenArray, Element, ElementRef} from 'react';

type Props = {
    children: ChildrenArray<Element<typeof Tab> | false>,
    className?: string,
    onSelect: (tabIndex: number) => void,
    selectedIndex: ?number,
    type: Type,
};

const DEBOUNCE_TIME = 200;

@observer
class Tabs extends React.Component<Props> {
    @observable tabsWidth: number = 0;
    @observable tabsContainerWrapperWidth: number = 0;
    @observable tabsContainerWidth: number = 0;

    @observable tabWidths: Map<number, number> = new Map();
    @observable tabRefs: Map<number, ?ElementRef<'li'>> = new Map();
    @observable dropdownOpen = false;
    @observable lastSelectedIndex: ?number;

    static Tab = Tab;

    resizeObserver: ?ResizeObserver;

    tabsRef: ?ElementRef<'div'>;
    tabsContainerWrapperRef: ?ElementRef<'div'>;
    tabsContainerRef: ?ElementRef<'ul'>;
    dropdownButtonRef: ?ElementRef<'button'>;

    componentDidMount() {
        this.setDimensions();

        this.resizeObserver = new ResizeObserver(
            debounce(this.setDimensions, DEBOUNCE_TIME)
        );

        if (this.tabsContainerWrapperRef) {
            this.resizeObserver.observe(this.tabsContainerWrapperRef);
        }

        if (this.tabsContainerRef) {
            this.resizeObserver.observe(this.tabsContainerRef);
        }
    }

    componentWillUnmount() {
        if (this.resizeObserver) {
            this.resizeObserver.disconnect();
        }
    }

    setTabsRef = (ref: ?ElementRef<'div'>) => {
        this.tabsRef = ref;
    };

    setTabsContainerWrapperRef = (ref: ?ElementRef<'div'>) => {
        this.tabsContainerWrapperRef = ref;
    };

    setTabsContainerRef = (ref: ?ElementRef<'ul'>) => {
        this.tabsContainerRef = ref;
    };

    setDropdownButtonRef = (ref: ?ElementRef<'button'>) => {
        this.dropdownButtonRef = ref;
    };

    @action setTabsWidth = () => {
        if (!this.tabsRef) {
            return;
        }

        const width = this.tabsRef.offsetWidth;
        const style = window.getComputedStyle(this.tabsRef);
        if (this.tabsWidth !== width) {
            this.tabsWidth = width - parseFloat(style.paddingLeft) - parseFloat(style.paddingRight);
        }
    };

    @action setTabsContainerWrapperWidth = () => {
        if (!this.tabsContainerWrapperRef) {
            return;
        }

        const width = this.tabsContainerWrapperRef.offsetWidth;
        if (this.tabsContainerWrapperWidth !== width) {
            this.tabsContainerWrapperWidth = width;
        }
    };

    @action setTabsContainerWidth = () => {
        if (!this.tabsContainerRef) {
            return;
        }

        const width = this.tabsContainerRef.offsetWidth;
        if (this.tabsContainerWidth !== width) {
            this.tabsContainerWidth = width;
        }
    };

    @action updateTabWidths = () => {
        this.tabRefs.forEach((ref, key) => {
            if (!ref) {
                return;
            }

            const width = ref.offsetWidth;
            if (this.tabWidths.get(key) !== width) {
                this.tabWidths.set(key, width);
            }
        });
    };

    setDimensions = () => {
        this.setTabsWidth();
        this.setTabsContainerWrapperWidth();
        this.setTabsContainerWidth();
        this.updateTabWidths();
    };

    @action setTabRef = (index: number, ref: ?ElementRef<'li'>) => {
        if (this.tabRefs.get(index) !== ref) {
            this.tabRefs.set(index, ref);
        }
    };

    @action handleDropdownToggle = () => {
        this.dropdownOpen = !this.dropdownOpen;
    };

    @action handleDropdownClose = () => {
        this.dropdownOpen = false;
    };

    changeTab = (selectedTabIndex: ?number) => {
        if (selectedTabIndex !== undefined && selectedTabIndex !== null) {
            this.props.onSelect(selectedTabIndex);
        }
    };

    handleTabClick = (selectedTabIndex: ?number) => {
        this.changeTab(selectedTabIndex);
    };

    @action handleCollapsedTabClick = (selectedTabIndex: number) => {
        this.dropdownOpen = false;
        this.lastSelectedIndex = selectedTabIndex;

        this.changeTab(selectedTabIndex);
    };

    isSelected(tabIndex: number) {
        return tabIndex === this.props.selectedIndex;
    }

    get childIndices(): number[] {
        const {children} = this.props;

        return React.Children.map(children, (child, index) => index);
    }

    @computed get visibleTabIndices(): number[] {
        if (this.tabsContainerWidth <= this.tabsWidth) {
            return this.childIndices;
        }

        const {selectedIndex} = this.props;

        let visibleWidth = 0;
        let visibleTabIndices: number[] = [];

        const childIndices = this.childIndices;

        // Sorts childIndices in it's natural order, except that the element with selectedIndex is positioned at the
        // first place and the element with lastSelectedIndex is positioned at the second place.
        // This ensures that those two elements will always be visible.
        childIndices.sort((a, b) => {
            if (a === selectedIndex) {
                return -1;
            }

            if (b === selectedIndex) {
                return 1;
            }

            if (a === this.lastSelectedIndex) {
                return -1;
            }

            if (b === this.lastSelectedIndex) {
                return 1;
            }

            return a - b;
        });

        for (const index of childIndices) {
            const nextWidth = this.tabWidths.get(index);

            if (undefined === nextWidth) {
                if (visibleTabIndices.length > 0) {
                    break;
                }

                return this.childIndices;
            }

            if (visibleWidth + nextWidth > this.tabsContainerWrapperWidth) {
                break;
            }

            visibleWidth += nextWidth;
            visibleTabIndices = [...visibleTabIndices, index];
        }

        // Since visibleTabIndices contains only the indices of the elements that can be fully shown and both,
        // selectedIndex and lastSelectedIndex are still positioned at the very beginning of the array, we need to reset
        // the sorting so the elements have the correct order again.
        visibleTabIndices.sort((a, b) => a - b);

        return visibleTabIndices;
    }

    @computed get collapsedTabIndices(): number[] {
        const visibleTabIndices = this.visibleTabIndices;

        return this.childIndices.filter((index) => !visibleTabIndices.includes(index));
    }

    @computed get hasCollapsedTabs(): boolean {
        return this.collapsedTabIndices.length > 0;
    }

    createTabItems(tabs: Array<Element<typeof Tab> | false>, indices: number[], hidden: boolean) {
        const {type} = this.props;

        return React.Children.map(tabs, (tab, localIndex) => {
            const index = indices[localIndex];

            if (!tab) {
                return null;
            }

            const selected = this.isSelected(index);

            return React.cloneElement(
                tab,
                {
                    ...tab.props,
                    hidden,
                    index,
                    selected,
                    type,
                    onClick: this.handleTabClick,
                    tabRef: this.setTabRef,
                }
            );
        });
    }

    createTabs = () => {
        const {children} = this.props;
        const {visibleTabIndices, collapsedTabIndices} = this;

        return [
            ...this.createTabItems(
                React.Children.toArray(children).filter(
                    (child, index) => visibleTabIndices.includes(index)
                ),
                visibleTabIndices,
                false
            ),
            ...this.createTabItems(
                React.Children.toArray(children).filter(
                    (child, index) => collapsedTabIndices.includes(index)
                ),
                collapsedTabIndices,
                true
            ),
        ];
    };

    createCollapsedTabItems(tabs: Array<Element<typeof Tab> | false>, indices: number[]) {
        return React.Children.map(tabs, (tab, localIndex) => {
            const index = indices[localIndex];

            if (!tab) {
                return null;
            }

            return (
                <CollapsedTab
                    index={index}
                    key={index}
                    onClick={this.handleCollapsedTabClick}
                >
                    {tab.props.children}
                </CollapsedTab>
            );
        });
    }

    createCollapsedTabs = () => {
        const {children} = this.props;
        const {collapsedTabIndices} = this;

        return this.createCollapsedTabItems(
            React.Children.toArray(children).filter((child, index) => collapsedTabIndices.includes(index)),
            collapsedTabIndices
        );
    };

    render() {
        const {
            type,
            className,
        } = this.props;

        const tabsClass = classNames(
            tabsStyles.tabs,
            tabsStyles[type],
            className
        );

        return (
            <div className={tabsClass} ref={this.setTabsRef}>
                <div className={tabsStyles.tabsContainerWrapper} ref={this.setTabsContainerWrapperRef}>
                    <ul className={tabsStyles.tabsContainer} ref={this.setTabsContainerRef}>
                        {this.createTabs()}
                    </ul>
                </div>

                {this.hasCollapsedTabs &&
                    <React.Fragment>
                        <button
                            className={tabsStyles.button}
                            onClick={this.handleDropdownToggle}
                            ref={this.setDropdownButtonRef}
                            type="button"
                        >
                            <Icon name="su-more-horizontal" />
                        </button>

                        <Popover
                            anchorElement={this.dropdownButtonRef || undefined}
                            onClose={this.handleDropdownClose}
                            open={this.dropdownOpen}
                        >
                            {
                                (setPopoverRef, styles) => (
                                    <div ref={setPopoverRef} style={styles}>
                                        <CollapsedTabList type={type}>
                                            {this.createCollapsedTabs()}
                                        </CollapsedTabList>
                                    </div>
                                )
                            }
                        </Popover>
                    </React.Fragment>
                }
            </div>
        );
    }
}

export default Tabs;
