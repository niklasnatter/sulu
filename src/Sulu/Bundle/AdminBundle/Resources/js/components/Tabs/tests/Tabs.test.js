/* eslint-disable flowtype/require-valid-file-annotation */
import {mount, render, shallow} from 'enzyme';
import React from 'react';
import debounce from 'debounce';
import Badge from '../../Badge/Badge';
import Tabs from '../Tabs.js';

jest.mock('debounce', () => jest.fn((callback) => callback));

window.ResizeObserver = jest.fn(function() {
    this.observe = jest.fn();
    this.disconnect = jest.fn();
});

Object.defineProperty(window, 'getComputedStyle', {
    value: () => ({
        paddingLeft: 20.0,
        paddingRight: 20.0,
    }),
});

test('Render a Tabs component with type root', () => {
    const changeSpy = jest.fn();

    expect(render(
        <Tabs onSelect={changeSpy} selectedIndex={null} type="root">
            <Tabs.Tab>Tab 1</Tabs.Tab>
            <Tabs.Tab>Tab 2</Tabs.Tab>
            <Tabs.Tab>Tab 3</Tabs.Tab>
        </Tabs>
    )).toMatchSnapshot();
});

test('Render a Tabs component with type nested', () => {
    const changeSpy = jest.fn();

    expect(render(
        <Tabs onSelect={changeSpy} selectedIndex={null} type="nested">
            <Tabs.Tab>Tab 1</Tabs.Tab>
            <Tabs.Tab>Tab 2</Tabs.Tab>
            <Tabs.Tab>Tab 3</Tabs.Tab>
        </Tabs>
    )).toMatchSnapshot();
});

test('Render a Tabs component with type inline', () => {
    const changeSpy = jest.fn();

    expect(render(
        <Tabs onSelect={changeSpy} selectedIndex={null} type="inline">
            <Tabs.Tab>Tab 1</Tabs.Tab>
            <Tabs.Tab>Tab 2</Tabs.Tab>
            <Tabs.Tab>Tab 3</Tabs.Tab>
        </Tabs>
    )).toMatchSnapshot();
});

test('Render a Tabs component with a selected tab and a badge', () => {
    const changeSpy = jest.fn();
    const selectedTabIndex = 0;

    expect(render(
        <Tabs onSelect={changeSpy} selectedIndex={selectedTabIndex} type="root">
            <Tabs.Tab>Tab 1</Tabs.Tab>
            <Tabs.Tab>Tab 2</Tabs.Tab>
            <Tabs.Tab badges={[<Badge key="badge1">1</Badge>, <Badge key="badge2">2</Badge>]}>Tab 3</Tabs.Tab>
        </Tabs>
    )).toMatchSnapshot();
});

test('Clicking on a Tab should call the onSelect handler', () => {
    const changeSpy = jest.fn();
    const selectedTabIndex = 0;

    const tabs = mount(
        <Tabs onSelect={changeSpy} selectedIndex={null}>
            <Tabs.Tab>Tab 1</Tabs.Tab>
            <Tabs.Tab>Tab 2</Tabs.Tab>
            <Tabs.Tab>Tab 3</Tabs.Tab>
        </Tabs>
    );

    tabs.find('.tab button').at(0).simulate('click');
    expect(changeSpy).toHaveBeenCalledWith(selectedTabIndex);
});

test('Clicking on several non- and collapsed tabs', () => {
    const resizeFunction = jest.fn();
    debounce.mockReturnValue(resizeFunction);

    const changeSpy = jest.fn();
    const selectedTabIndex = 0;

    const tabs = shallow(
        <Tabs onSelect={changeSpy} selectedIndex={selectedTabIndex}>
            <Tabs.Tab>Tab 1</Tabs.Tab>
            <Tabs.Tab>Tab 2</Tabs.Tab>
            <Tabs.Tab>Tab 3</Tabs.Tab>
            <Tabs.Tab>Tab 4</Tabs.Tab>
            <Tabs.Tab>Tab 5</Tabs.Tab>
            <Tabs.Tab>Tab 6</Tabs.Tab>
            <Tabs.Tab>Tab 7</Tabs.Tab>
            <Tabs.Tab>Tab 8</Tabs.Tab>
            <Tabs.Tab>Tab 9</Tabs.Tab>
            <Tabs.Tab>Tab 10</Tabs.Tab>
        </Tabs>
    );

    tabs.instance().tabsRef = {
        offsetWidth: 129,
    };

    tabs.instance().tabsContainerWrapperRef = {
        offsetWidth: 55,
    };

    tabs.instance().tabsContainerRef = {
        offsetWidth: 100,
    };

    tabs.instance().setDimensions();
    tabs.instance().tabWidths = new Map([
        [0, 10],
        [1, 10],
        [2, 10],
        [3, 10],
        [4, 10],
        [5, 10],
        [6, 10],
        [7, 10],
        [8, 10],
        [9, 10],
    ]);

    tabs.update();

    // Initial state
    expect(tabs.instance().visibleTabIndices).toEqual([0, 1, 2, 3, 4]);
    expect(tabs.instance().collapsedTabIndices).toEqual([5, 6, 7, 8, 9]);

    // Click on already visible tab
    tabs.instance().handleTabClick(4);
    expect(changeSpy).toBeCalledWith(4);
    tabs.setProps({selectedIndex: 4});

    expect(tabs.instance().visibleTabIndices).toEqual([0, 1, 2, 3, 4]);
    expect(tabs.instance().collapsedTabIndices).toEqual([5, 6, 7, 8, 9]);

    // Click on hidden tab
    tabs.instance().handleCollapsedTabClick(6);
    expect(changeSpy).toBeCalledWith(6);
    tabs.setProps({selectedIndex: 6});

    expect(tabs.instance().visibleTabIndices).toEqual([0, 1, 2, 3, 6]);
    expect(tabs.instance().collapsedTabIndices).toEqual([4, 5, 7, 8, 9]);

    // Click on another hidden tab
    tabs.instance().handleCollapsedTabClick(8);
    expect(changeSpy).toBeCalledWith(8);
    tabs.setProps({selectedIndex: 8});

    expect(tabs.instance().visibleTabIndices).toEqual([0, 1, 2, 3, 8]);
    expect(tabs.instance().collapsedTabIndices).toEqual([4, 5, 6, 7, 9]);

    // Click on visible tab again
    tabs.instance().handleTabClick(2);
    expect(changeSpy).toBeCalledWith(2);
    tabs.setProps({selectedIndex: 2});

    expect(tabs.instance().visibleTabIndices).toEqual([0, 1, 2, 3, 8]);
    expect(tabs.instance().collapsedTabIndices).toEqual([4, 5, 6, 7, 9]);

    // Click again on another visible tab
    tabs.instance().handleTabClick(3);
    expect(changeSpy).toBeCalledWith(3);
    tabs.setProps({selectedIndex: 3});

    expect(tabs.instance().visibleTabIndices).toEqual([0, 1, 2, 3, 8]);
    expect(tabs.instance().collapsedTabIndices).toEqual([4, 5, 6, 7, 9]);
});

test('ResizeObserver.disconnect should be called before component unmount', () => {
    const changeSpy = jest.fn();
    const selectedTabIndex = 0;

    const tabs = shallow(
        <Tabs onSelect={changeSpy} selectedIndex={selectedTabIndex}>
            <Tabs.Tab>Tab 1</Tabs.Tab>
            <Tabs.Tab>Tab 2</Tabs.Tab>
            <Tabs.Tab>Tab 3</Tabs.Tab>
            <Tabs.Tab>Tab 4</Tabs.Tab>
            <Tabs.Tab>Tab 5</Tabs.Tab>
            <Tabs.Tab>Tab 6</Tabs.Tab>
            <Tabs.Tab>Tab 7</Tabs.Tab>
            <Tabs.Tab>Tab 8</Tabs.Tab>
            <Tabs.Tab>Tab 9</Tabs.Tab>
            <Tabs.Tab>Tab 10</Tabs.Tab>
        </Tabs>
    );

    tabs.instance().componentWillUnmount();
    expect(ResizeObserver.mock.instances[0].disconnect).toBeCalled();
});
