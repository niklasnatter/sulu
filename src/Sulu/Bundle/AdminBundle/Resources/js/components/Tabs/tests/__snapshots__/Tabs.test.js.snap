// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a Tabs component with a selected tab and a badge 1`] = `
<div
  class="tabs root"
>
  <div
    class="tabsContainerWrapper"
  >
    <ul
      class="tabsContainer"
    >
      <li
        class="tab root selected"
      >
        <button
          disabled=""
          title="Tab 1"
          type="button"
        >
          Tab 1
        </button>
      </li>
      <li
        class="tab root"
      >
        <button
          title="Tab 2"
          type="button"
        >
          Tab 2
        </button>
      </li>
      <li
        class="tab root"
      >
        <button
          title="Tab 3"
          type="button"
        >
          Tab 3
          <div
            class="badges"
          >
            <div
              class="badge"
            >
              1
            </div>
            <div
              class="badge"
            >
              2
            </div>
          </div>
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Render a Tabs component with type inline 1`] = `
<div
  class="tabs inline"
>
  <div
    class="tabsContainerWrapper"
  >
    <ul
      class="tabsContainer"
    >
      <li
        class="tab inline"
      >
        <button
          title="Tab 1"
          type="button"
        >
          Tab 1
        </button>
      </li>
      <li
        class="tab inline"
      >
        <button
          title="Tab 2"
          type="button"
        >
          Tab 2
        </button>
      </li>
      <li
        class="tab inline"
      >
        <button
          title="Tab 3"
          type="button"
        >
          Tab 3
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Render a Tabs component with type nested 1`] = `
<div
  class="tabs nested"
>
  <div
    class="tabsContainerWrapper"
  >
    <ul
      class="tabsContainer"
    >
      <li
        class="tab nested"
      >
        <button
          title="Tab 1"
          type="button"
        >
          Tab 1
        </button>
      </li>
      <li
        class="tab nested"
      >
        <button
          title="Tab 2"
          type="button"
        >
          Tab 2
        </button>
      </li>
      <li
        class="tab nested"
      >
        <button
          title="Tab 3"
          type="button"
        >
          Tab 3
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Render a Tabs component with type root 1`] = `
<div
  class="tabs root"
>
  <div
    class="tabsContainerWrapper"
  >
    <ul
      class="tabsContainer"
    >
      <li
        class="tab root"
      >
        <button
          title="Tab 1"
          type="button"
        >
          Tab 1
        </button>
      </li>
      <li
        class="tab root"
      >
        <button
          title="Tab 2"
          type="button"
        >
          Tab 2
        </button>
      </li>
      <li
        class="tab root"
      >
        <button
          title="Tab 3"
          type="button"
        >
          Tab 3
        </button>
      </li>
    </ul>
  </div>
</div>
`;
